authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1015 @@
+  }
+
+  /**
+   * 1. Find the policy base securable object. <br>
+   * 2. If the policy is exist, then user new securable object's privilege to update. <br>
+   * 3. If the policy is not exist return false. <br>
+   */
+  private boolean doUpdateSecurableObject(
+      String roleName, RoleChange.UpdateSecurableObject change) {
+    RangerPolicy policy = findManagedPolicy(change.getSecurableObject());
+
+    if (policy != null) {
+      removePolicyItem(policy, roleName, change.getSecurableObject());
+      addPolicyItem(policy, roleName, change.getNewSecurableObject());
+      try {
+        if (policy.getId() == null) {
+          rangerClient.createPolicy(policy);
+        } else {
+          rangerClient.updatePolicy(policy.getId(), policy);
+        }
+      } catch (RangerServiceException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      LOG.warn(
+          "Cannot find the policy({}) for the securable object({})!",
+          roleName,
+          change.getSecurableObject().fullName());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Add policy item access items base the securable object's privileges. <br>
+   * We didn't clean the policy items, because one Ranger policy maybe contain multiple Gravitino
+   * securable objects. <br>
+   */
+  private void addPolicyItem(
+      RangerPolicy policy, String roleName, SecurableObject securableObject) {
+    // First check the privilege if support in the Ranger Hive
+    checkSecurableObject(securableObject);
+
+    // Add the policy items by the securable object's privileges
+    securableObject
+        .privileges()
+        .forEach(
+            gravitinoPrivilege -> {
+              // Translate the Gravitino privilege to mapped Ranger privilege
+              translatePrivilege(gravitinoPrivilege.name())
+                  .forEach(
+                      mappedPrivilege -> {
+                        // Find the policy item that match access and role
+                        boolean exist =
+                            policy.getPolicyItems().stream()
+                                .anyMatch(
+                                    policyItem -> {
+                                      return policyItem.getRoles().contains(roleName)
+                                          && policyItem.getAccesses().stream()
+                                              .anyMatch(
+                                                  access ->
+                                                      access.getType().equals(mappedPrivilege));
+                                    });
+                        if (exist) {
+                          return;
+                        }
+
+                        RangerPolicy.RangerPolicyItem policyItem =
+                            new RangerPolicy.RangerPolicyItem();
+                        RangerPolicy.RangerPolicyItemAccess access =
+                            new RangerPolicy.RangerPolicyItemAccess();
```

Comment:
If a policy item already exists with a matching accessType and mappedPrivilege, should we update this policy  item instead of continuing to create a new policy item? 

We should ensure that a privilege has only one policy item to improve maintainability.

Reply:
Yes, Currently one privilege has only one policy item.

Reply:
But the current logic doesn't seem to be like that.

If the condition is true, we should update this policyItem.
```
policyItem.getAccesses().stream().anyMatch(access ->access.getType().equals(mappedPrivilege))
```